### DISCUSSION TITLE
Is there a way to share cookies between different domains using next.js api?

### DISCUSSION THREAD

**Discussion by hoo00nn** [vercel/next.js#37904] [Help] [answered]
Hello, next.js team

You are using Next.js well, but you have encountered one problem.

When sending a request to the Next.js API from another domain, Next.js is not receiving cookies.

The client set it to withCredentials: true and next.js set it to credentials: true using the cors package.

Requests between cross-origin are normal, but cookies cannot be imported.

I suspect the google same-site policy issue. Can I set same-site to none in next.js?

Help me.

**Comment by icyJoseph** [comment, marked as answer]
If the "another domain" cookies are not set with same-site: none, then your API routes have no say in the matter.

It'd be a huge security risk if third parties could just access a domain cookies through a simple request. I think this used to be the default behavior a few years prior. Not anymore.

Who ever is setting the cookie in "another domain" has to set them with SameSite None and Secure. If it is controlled by you, you can easily add those options when setting the cookie. If its a 3rd party, you'd have to ask them to do it, which they'll probably refuse, for security concerns.

This is not a google policy thing, is how browsers in general work, some are even more aggressive and prohibit any type of cookie sharing. SSO solutions often have to fight this with very elaborated schemes.

**Reply by hoo00nn** [reply]
Thank you. It was very helpful.

**Comment by IlirEdis** [comment]
@hoo00nn did you succeed with this? If so, do you have any sample code how you did this? 
I'm struggling to update cart in another domain.

**Reply by mok419** [reply]
I tried and failed today 😭 any success?

**Reply by Ryine** [reply]
fixed it now？

**Reply by IlirEdis** [reply]
No, i left that project.

**Reply by mok419** [reply]
> fixed it now？

Check these out, I think they patched it in 2024 and are phasing it out. That was the case for what I was trying!

https://github.com/GoogleChromeLabs/samesite-examples/issues/53#issue-2124157381

https://github.com/balazsorban44/auth-poc-next/issues/1#issue-2117783837
